sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { WebResourceLike } from "@azure/core-http";
+import {
+  createDefaultHttpClient,
+  createPipelineRequest,
+  HttpClient,
+  PipelinePolicy,
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest
+} from "@azure/core-rest-pipeline";
+
+const paths = {
+  playback: "/playback",
+  record: "/record",
+  start: "/start",
+  stop: "/stop"
+};
+
+/**
+ * Helper class to manage the recording state to make sure the proxy-tool is not flooded with unintended requests.
+ *
+ * => then start
+ * => run the runAsync
+ * => stop record
+ * => start playback
+ * => stop playback
+ */
+export class RecordingStateManager {
+  public state: "started" | "stopped" | undefined;
+
+  /**
+   * validateState
+   */
+  public validateState(currentFlow: "starting" | "stopping") {
+    if (currentFlow === "starting") {
+      if (this.state === "started") {
+        throw new Error("Recorder Error: Already started, should not have called start again.");
+      }
+    }
+    if (currentFlow === "stopping") {
+      if (this.state === "stopped") {
+        throw new Error("Recorder Error: Already stopped, should not have called stop again.");
+      }
+      if (this.state !== "started") {
+        throw new Error("Recorder Error: Please start before calling stop.");
+      }
+    }
+    if (currentFlow === "starting") {
+      if (this.state !== "stopped" && this.state !== undefined) {
+        throw new Error("Recorder Error: Please stop before calling start.");
+      }
+    }
+    if (currentFlow === "stopping") {
+      if (this.state !== "started") {
+        throw new Error("Recorder Error: Please start before calling stop.");
+      }
+    }
+  }
+
+  /**
+   * setState
+   */
+  public setState(state: "started" | "stopped") {
+    this.state = state;
+  }
+}
+
+export class TestProxyHttpClient {
+  private url: string;
+  public recordingId?: string;
+  public mode: string;
+  public httpClient: HttpClient;
+  private sessionFile: string;
+  private playback: boolean;
+  private stateManager = new RecordingStateManager();
+
+  constructor(sessionFile: string, playback: boolean) {
+    this.sessionFile = sessionFile;
+    this.playback = playback;
+    this.url = "http://localhost:5000";
```

Comment:
`go` and `python` both have SSL enabled. Let's chat at some point after this and figure out what it takes to make it work for JS. We'll update the doc [here](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/documentation/trusting-cert-per-language.md#js) and leave it up to you folks to see if it's important to ya :)

Reply:
Sure, logged an issue #17070.
Not a blocker right now.